mockgen/version.1.12.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2010 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// +build go1.12
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"runtime/debug"
+)
+
+func printVersion() {
+	if bi, exists := debug.ReadBuildInfo(); exists {
+		fmt.Println(bi.Main.Version)
```

Comment:
What does this actually print out today if you run it on go1.12+?

Reply:
As soon as we release 1.3.2, the `-version` flag should report `v1.3.2` if `go get` is invoked with `@latest`. I think there is no way to test this with `mockgen` itself now. But I tested the `debug.ReadBuildInfo` in my other test project. 

One edge case is when somebody invokes the `go get` with `@master`. But it does behave correctly. With `@master` the resulting version should be something like: `v1.3.2-0.20191220092011-827cab0badc9`, in other words if the `go get` is requsted to download an unreleased version, the module support adds commit SHA to the version string to disambiguate the version.

There are multiple situations when `-version` is not going to be available (but I think that is fine):
 - when building/running the mockgen module locally the `-version` is `(devel)`
 - when running with Golang older then 1.12 (as seen in the changes in this PR)